src/gui/torrentcreatordlg.cpp

```diff
@@ -28,254 +28,245 @@
  * Contact : chris@qbittorrent.org
  */
 
+#include "torrentcreatordlg.h"
+
+#include <QCloseEvent>
 #include <QDebug>
 #include <QFileDialog>
 #include <QMessageBox>
+#include <QMimeData>
+#include <QUrl>
 
-#include "torrentcreatordlg.h"
-#include "base/utils/fs.h"
-#include "base/utils/misc.h"
-#include "base/preferences.h"
-#include "guiiconprovider.h"
+#include "base/global.h"
 #include "base/bittorrent/session.h"
 #include "base/bittorrent/torrentinfo.h"
 #include "base/bittorrent/torrentcreatorthread.h"
+#include "base/settingsstorage.h"
+#include "base/utils/fs.h"
+#include "guiiconprovider.h"
 
-const uint NB_PIECES_MIN = 1200;
-const uint NB_PIECES_MAX = 2200;
+#include "ui_torrentcreatordlg.h"
 
-TorrentCreatorDlg::TorrentCreatorDlg(QWidget *parent)
-    : QDialog(parent)
-    , m_creatorThread(0)
+namespace
 {
-    setupUi(this);
-    // Icons
-    addFile_button->setIcon(GuiIconProvider::instance()->getIcon("document-new"));
-    addFolder_button->setIcon(GuiIconProvider::instance()->getIcon("folder-new"));
-    createButton->setIcon(GuiIconProvider::instance()->getIcon("document-save"));
-    cancelButton->setIcon(GuiIconProvider::instance()->getIcon("dialog-cancel"));
+#define SETTINGS_KEY(name) "TorrentCreator/" name
+    StorableValue<QSize> storeDialogSize(SETTINGS_KEY("Dimension"));
```

Comment:
Oh, I'm sorry, we don't use pointers here (and it would be extremely inconvenient), so Forward Declaration is not suitable here.
But in this case PIMPL is overkill too.